src/fastmcp/prompts/prompt.py

```
@@ -199,6 +199,13 @@ def from_function(
             if param.kind == inspect.Parameter.VAR_KEYWORD:
                 raise ValueError("Functions with **kwargs are not supported as prompts")
 
+        # Validate that task=True requires async functions
+        if task and not inspect.iscoroutinefunction(fn):
+            raise ValueError(
+                f"Prompt '{func_name}' uses a sync function but has task=True. "
+                "Background tasks require async functions. Set task=False to disable."
+            )
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)